adapters/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/uri/BaseUri.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.raml.jms.uri;
+
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+
+/**
+ * Parses Raml base uri and exposes it's parts through accessor methods
+ */
+public class BaseUri {
+    private static final Pattern MESSAGING_BASE_URI_PATTERN
+            = Pattern.compile("message://(event|command|query)/(api|controller|handler|listener|processor)/\\S+/(\\S+)");
```

Comment:
This needs to support 'view' in addition to api|controller|handler|listener|processor